src/backend/core/roles/decorator.py

```diff
@@ -0,0 +1,47 @@
+from functools import wraps
+from rest_framework.response import Response
+
+from user.models import User
+
+from .roles_manager import RolesManager
+
+
+def check_user_access(wanted_permissions, user_id):
+    user = User.objects.get(email=user_id)
+    user_groups = user.groups.all()
+    user_permissions = user.user_permissions.all()
+    user_permissions_set = set()
+    for perm in user_permissions:
+        user_permissions_set.add(
+            RolesManager.DjangoPermissionToManagerPermission(perm).name
+        )
+    for group in user_groups:
+        for perm in group.permissions.all():
+            user_permissions_set.add(
+                RolesManager.DjangoPermissionToManagerPermission(perm).name
+            )
+
+    if wanted_permissions.issubset(user_permissions_set):
+        return True
+    return False
+
+
+def check_user_access_decorator(permissions):
```

Comment:
It would be nice to add doc comment to the decorator (since I assume we will then heavily use it for authorization)

Reply:
Agreed, I will add that.